modules/plugins/plugins-compat.php

```diff
@@ -124,6 +124,11 @@ public function plugins_loaded() {
 		if ( function_exists( 'custom_post_widget_plugin_init' ) && class_exists( 'PLL_Content_Blocks' ) ) {
 			add_action( 'pll_init', array( $this->content_blocks = new PLL_Content_Blocks(), 'init' ) );
 		}
+
+		// AMP
+		if ( function_exists( 'amp_get_slug' ) ) {
+			add_filter( 'pll_front_page_ignored_query_vars', array( $this, 'add_amp_to_front_page_ignored_query_vars' ) );
```

Comment:
Since you propose to add a filter hook, wouldn't it make sense to register this filter callback that triggers one of AMP functions, inside the AMP plugin?

Reply:
This is a great question!

The problem for the AMP plugin is that adding all the plugin compatibility layers is not really scalable and difficult to maintain. That's why the AMP plugin has always been trying to advocate to other plugins and themes in the ecosystem to handle that in their projects; with the AMP plugin serving as the "framework" to support that.

So I'd argue to keep this in Polylang. If you ever change something in this logic, people wouldn't have to wait for an accompanying change in the AMP plugin but could instead just update Polylang alone and it would _just work_. Does that  makes sense for you?

cc @westonruter in case you want to add something

Reply:
Yeah, that sounds right. Also, Polylang is already adding compatibility for other plugins, so it seems to fit nicely here.